tests/BookReader.test.js

```diff
@@ -7,3 +10,51 @@ test('clamp function returns Math.min(Math.max(value, min), max)', () => {
 test('calculate a percentage suitable for CSS', () => {
     expect(BookReader.util.cssPercentage(2,1)).toEqual(200+'%');
 });
+
+test('escapeHTML function which replaces the string', () => {
+    expect(BookReader.util.escapeHTML('Me & You')).toEqual('Me &amp; You');
+    expect(BookReader.util.escapeHTML('Me > You')).toEqual('Me &gt; You');
+    expect(BookReader.util.escapeHTML('Me < You')).toEqual('Me &lt; You');
+    expect(BookReader.util.escapeHTML('Me " You')).toEqual('Me &quot; You');
+});
+
+test('Decodes a URI component and converts + to emptyStr', () => {
+    expect(BookReader.util.decodeURIComponentPlus("https%3A%2F%2Farchive.org%2Fskr+")).toEqual("https://archive.org/skr ");
+    expect(BookReader.util.decodeURIComponentPlus("%3Fx%3D+test")).toEqual("?x= test");
+});
+
+test('Encodes a URI component and converts emptyStr to +', () => {
+    expect(BookReader.util.encodeURIComponentPlus("?x=test ")).toEqual("%3Fx%3Dtest+");
+    expect(BookReader.util.encodeURIComponentPlus("ABC abc 123")).toEqual("ABC+abc+123");
+});
+
+let clock;
+
+beforeEach(() => {
+  clock = sinon.useFakeTimers();
+});
+
+afterEach(() => {
+  clock.restore();
+});
+
+test('testing debounce', () => {
```

Comment:
What are you testing in this test?

Reply:
BookReader.util.debounce() function present in BookReader.js

Reply:
I have missed the BookReader.util part.. I will add it.